MDANSE/Src/MDANSE/Mathematics/Signal.py

```diff
@@ -292,3 +298,643 @@ def get_spectrum(signal, window=None, timeStep=1.0, axis=0, fft="fft"):
+    dw = pre_w[1] - pre_w[0]
+    pre_amplitudes = (2 * (np.abs(pre_filter_freqs["h"])) / N)[:np.int32(N/2)]
+
+    import matplotlib.pyplot as plt
+
+    plt.figure(figsize=(10, 6))
+    plt.plot(pre_w, pre_amplitudes, label="pre-filter")
+    plt.show()
+
+    f = filter_class(**{
+        "n_steps": 10000,
+        "time_step_ps": 0.005,
+        "order": 2,
+        "attenuation_type": "highpass",
+        "cutoff_freq": 25.0,
+        "max_ripple": 1.0
+    })
+
+    data1 = f.apply(data)
+
+    post_filter_freqs = {
+        "h": fftpack.fft(data1),
+        "w": fftpack.fftfreq(N, d=1/fs)
+    }
+
+    post_w = post_filter_freqs["w"][:np.int32(N/2)]
+    post_amplitudes = (2 * (np.abs(post_filter_freqs["h"])) / N)[:np.int32(N/2)]
+
+    plt.figure(figsize=(10, 6))
+    plt.plot(post_w, post_amplitudes, label="post-filter")
+    plt.show()
+
+    # w0 = 2 * np.pi * 1.5
+    # a0 = 20.0
+    #
+    # w1 = 2 * np.pi * 60.0
+    # a1 = 5.0
+    #
+    # N = 10000
+    # t = np.linspace(0, 20, N)
+    # fs = (t[1] - t[0])**(-1)
+    # x0 = a0 * np.sin(w0*t) + a1 * np.sin(w1*t)
+    #
+    # pre_filter_freqs = {
+    #     "h": fftpack.fft(x0),
+    #     "w": fftpack.fftfreq(N, d=1/fs)
+    # }
+    #
+    # pre_w = pre_filter_freqs["w"][:np.int32(N/2)]
+    # dw = pre_w[1] - pre_w[0]
+    # pre_amplitudes = (2 * (np.abs(pre_filter_freqs["h"])) / N)[:np.int32(N/2)]
+
+    import matplotlib.pyplot as plt
+    plt.figure(figsize=(10, 6))
+    # plt.plot(pre_w, pre_amplitudes, label="pre-filter")
+    plt.show()
+
+    #import matplotlib.pyplot as plt
+    #plt.figure(figsize=(10, 6))
+    #plt.plot(t, x, label="pre-filter")
+    #plt.show()
+
+    #f = filter_class(**{
+    #    "attenuation_type": "lowpass",
+    #    "order": 1,
+    #    "cutoff_freq": 60.0,
+    #    "time_step_ps": 1/fs,
+    #    "n_steps": N}
+    #)
+
+    #x1 = f.apply(x0)
+
+    #post_filter_freqs = {
+    #    "h": fftpack.fft(x1),
+    #    "w": fftpack.fftfreq(N, d=1/fs)
+    #}
+
+#    post_w = post_filter_freqs["w"][:np.int32(N/2)]
+#    post_amplitudes = (2 * (np.abs(post_filter_freqs["h"])) / N)[:np.int32(N/2)]
+
+    import matplotlib.pyplot as plt
+    plt.figure(figsize=(10, 6))
+    #plt.plot(post_w, post_amplitudes, label="post-filter")
+    plt.show()
+
+    #s = f.__str__()
+    #r = f.__repr__()
+
+
+
+    #plt.figure(figsize=(10, 6))
+    #plt.plot(t, f.apply(x), label="post-filter")
+    #plt.show()
+
+    #string_representation = [
+    #    f"Trajectory filter of type {1.0} implemented with the following parameters:\n\n",
+    #    f"  # sample_freq\n  Reciprocal of the molecular dynamics time step, in picohertz\n      {2.0}\n\n",
+    #    f"  # Sample frequency\n  Reciprocal of the molecular dynamics time step in picohertz\n      {3.0}\n\n",
+    #]
+
+    #string_representation.append(f"  # Cutoff frequency\n  Reciprocal of the molecular dynamics time step in picohertz\n      {4.0}\n\n")
+
+    #print("".join(string_representation))
```

Comment:
Can you move or remove this? I don't think this belongs here.

Reply:
Yeah, I will make sure this is gone before anything is merged - it's just a rough script for testing